packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -130,7 +130,8 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
     }
 
     public get clientDetails(): IClientDetails {
-        return this.dataStoreContext.containerRuntime.clientDetails;
+        // back-compat 0.38 - clientDetails is added to IFluidDataStoreContext in 0.38.
+        return this.dataStoreContext.clientDetails ?? this.dataStoreContext.containerRuntime.clientDetails;
```

Comment:
I'm not sure you want [hasOwnProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty), as it doesn't walk the prototype chain, in might be a better fit, so `clientDetails in this.dataStoreContext ? this.dataStoreContext.clientDetails : this.dataStoreContext.containerRuntime.clientDetails`

Reply:
Good point. But even with `in` there are typescript compiler error that I don't think is worth handling. I changed the tests to have an empty container runtime instead.